flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/agg/WindowAggregateTest.scala

```diff
@@ -496,6 +518,29 @@ class WindowAggregateTest(aggPhaseEnforcer: AggregatePhaseStrategy) extends Tabl
     util.verifyRelPlan(sql)
   }
 
+  @TestTemplate
+  def testCumulateWithNegativeParam(): Unit = {
+    val sql =
+      """
+        |SELECT
+        |   a,
+        |   window_start,
+        |   window_end,
+        |   count(*),
+        |   sum(d),
+        |   max(d) filter (where b > 1000),
+        |   weightedAvg(b, e) AS wAvg,
+        |   count(distinct c) AS uv
+        |FROM TABLE(
+        |  CUMULATE(TABLE MyTable, DESCRIPTOR(rowtime), INTERVAL '-10' MINUTE, INTERVAL '1' HOUR))
```

Comment:
Nit: what about verifying negative size?

Reply:
I would cover it soon.